### ISSUE TITLE
Heads up - new clone field for ACF Pro

### ISSUE THREAD

**Issue by Hube2** [Hube2/acf-reusable-field-group-field#29]
Elliot has begun working on a "Clone" field for ACF Pro. The new field will do basically what this add on does. I doubt that it will be compatible with or save data the way that this add on does.

Just wanted to let everyone know. 

I've been grateful for the support that this add on has received. 

I won't be deleting this add on, I know there are people using it and since the new field will not be compatible it would cause a lot of work to switch over to the new one. Also, it may be a while before all the bug are worked out of it and who know, this one may still be able to do things the built in field does not.

**Comment by Hube2** [comment]
I have added some information to the Readme about the new clone field. I would suggest that anyone that is not already using this plugin does not use it if it is at all possible that you can delay your development, or at least the parts that would use this plugin, until the new version of ACF is released.

Like I said, I won't be abandoning this plugin for those that are using it, but I don't think it would be the best idea to use this plugin if the new feature of ACF will do what you need done.

I will also provide assistance here if you have questions about the possibility of moving from this plugin to the new clone field in ACF.

I would greatly appreciate anyone that is able to move the data from this plugin to the new field in ACF to post there experience and how-to's either here or anywhere else and posting a link to it here.

**Comment by pxwee5** [comment]
Downloaded the beta and tested it. Does not work as it says. Prefixes aren't included. 

A lot of work pending on the new clone field.

**Comment by Hube2** [comment]
I put a bold noted at the top of README.md. No, the new clone field will not be compatible with this plugin. If you need to or want to switch there is also some advice there about how I think it can be accomplished. If there are questions I can try to answer them.

**Comment by pxwee5** [comment]
Yes I understand, I downloaded the beta (ACF 5.4) and tested it. 
NOTE: I replaced the entire 5.3.8 folder with 5.4beta without this plugin.

I didn't go through the codes but I think the clone is largely still under development. Can you confirm? 

I needed the feature on Clone 5.4 where I can add prefix names to the cloned fields. The clone field in 5.3.8 doesn't offer that.

**Comment by Hube2** [comment]
To be honest, I don't know exactly where development stands on the clone field, I've been too busy with work to keep up with it or do much testing.

**Comment by pxwee5** [comment]
Noted and thanks Hube2 for the update. :+1:

**Comment by Hube2** [comment]
Just thought I'd keep people updated. Elliot is still working out some bugs in the new clone field, most of which I solved by adding the whole field name and field key prefixing stuff to this plugin. Conditional logic problems and correctly saving and getting values. I will be interested in seeing how he solves the problems I could not without the complicated prefixing.

**Comment by Hube2** [comment]
ACF 5.4 has been released. I will attempt to give it a good testing to see what the differences between the clone field and this plugin are, but it might be a while before I have the time. I believe that the majority of the differences will be in the way that conditional logic will work with cloned fields and the way I implemented conditional logic. 

If anyone else has the time to do any testing and report back here what the differences are it would be greatly appreciated. 

Thanks,
~JH

**Comment by rg-** [comment]
Hi Hube2, i´m doing that also, testing with acf 5.4. For now seems like you can use both things and should not confilct. I didnt test that about conditional things.. i will take in cosideration for test/debug.

**Comment by rg-** [comment]
Currenty i have a field group that has a selector that condition other things behind, one of those are settings/items for a slider. That "slider" is in fact a reusable (only) group. I don´t see any problem with the conditional logic, back or front end working as before update.

**Comment by rg-** [comment]
mmm, unless the thing you commented is about conditional logics INSIDE a reusable field.... let´s see.

**Comment by Hube2** [comment]
So, I did some preliminary testing and here are some of the differences.

I may be easier to create conflicts in field names. ACF has the ability to turn on field name prefixing where this plugin makes you turn it off instead, erring on the side of caution.

It could being a bit easier with the clone field to create some conflicts.

The biggest difference between the clone field and this plugin is the conditional logic. This is not a bad thing, it just depends on what it is you want to do.

Basically the clone field does not have any conditional logic. You cannot for example create a clone field that includes an entire group or a field from another group and set the display of these fields based on another field in the group you're cloning into. To do this you need to add your own JS to deal with it or you can modify the field by using an acf/load_field filter to set the conditional logic yourself.

In this plugin conditional logic is applied to fields in group that is resused automatically.

There will not be a problem with duplicate field keys in the clone field. The cloned fields are given a new field key and the original field key is stored in the a new '_clone' index.

You can read more about it over on the ACF site, there is new documentation https://www.advancedcustomfields.com/resources/clone/

With the clone field you can also clone individual fields, and this plugin will only reuse entire groups. Honestly, I'm not sure I'd every use that feature, but I'm sure that some will.

As a result of this, I won't be looking to stop maintaining this plugin. I think there are enough differences between this and the clone field to make it worth keeping around.

I have not tested how this plugin deals with cloned fields though and I don't know if you can mix them. If anyone runs into any limitations or bugs with doing this, open a new issue and I'll look into it.

**Comment by rg-** [comment]
I think the same, your addon is very usefull, keep on that, i´ll keep on testing and reporting if any thing...

:)

**Comment by Hube2** [comment]
For anyone that's still following this. I've been giving the clone field a serious workout and it appears to be buggy in a few places and like I've said before I think it's mostly to do with problems caused by not prefixing field keys. Needless to say, I will not be discontinuing this plugin any time soon... I have to have it for a current project that can't wait until the clone field is free of bugs.
